helix-rest/src/main/java/org/apache/helix/rest/metadatastore/accessor/ZkRoutingDataWriter.java

```diff
@@ -209,8 +214,20 @@ public synchronized boolean setRoutingData(Map<String, List<String>> routingData
       return true;
     }
 
-    // TODO: Forward the request to leader
-    return true;
+    String leaderHostName = _leaderElection.getCurrentLeaderInfo().getId();
+    String url = leaderHostName + constructUrlSuffix(
+        MetadataStoreRoutingConstants.MSDS_GET_ALL_ROUTING_DATA_ENDPOINT);
+    HttpPut httpPut = new HttpPut(url);
+    String routingDataJsonString;
+    try {
+      routingDataJsonString = new ObjectMapper().writeValueAsString(routingData);
```

Comment:
Could we create a constant variable ObjectMapper instead of creating one every time this endpoint is called?

private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();

and you could use this throughout the class :)

Reply:
Sure.